scripts/west_commands/runners/nrf_common.py

```diff
@@ -307,6 +315,19 @@ def program_hex(self):
                 self.build_conf.getboolean('CONFIG_SOC_NRF54H20_ENGB_CPURAD') or
                 self.build_conf.getboolean('CONFIG_SOC_NRF9280_CPURAD')
             )
+            generated_uicr = self.build_conf.getboolean('CONFIG_NRF_REGTOOL_GENERATE_UICR')
+
+            if cpuapp:
+                core = 'NRFDL_DEVICE_CORE_APPLICATION'
+            elif cpurad:
+                core = 'NRFDL_DEVICE_CORE_NETWORK'
+
+            if generated_uicr and not self.hex_get_uicrs().get(core):
```

Comment:
As a follow-up, would it be possible for the script to check the UICRs for resource allocation conflicts? If there is a conflict, then the device won't boot and there won't be any hint to the user that there's an allocation conflict.

Reply:
Yes, this is an upcoming feature.